app/javascript/app/service-provider-form.js

```diff
@@ -1,52 +1,68 @@
-  const ialLevel = $('#service_provider_ial');
-  const samlFields = $('.saml-fields');
-  const oidcFields = $('.oidc-fields');
-  const ialAttributesCheckboxes = $('.ial-attr-wrapper');
-  const fileContainer = $('#certificate-container');
-  const logoInput = $('.input-file');
-  const pemInputMessage = $('.js-pem-input-error-message');
-  const pemInput = $('.js-pem-input');
-  const redirectURI = $('#add-redirect-uri-field');
-  const failureToProofURL = $('.service_provider_failure_to_proof_url');
-  const returnToSpUrl = document.querySelector('#service_provider_return_to_sp_url');
-
+  const ialLevel = document.querySelector('#service_provider_ial');
+  const ialAttributesCheckboxes = document.querySelectorAll('.ial-attr-wrapper');
+  const failureToProofURL = document.querySelector('.service_provider_failure_to_proof_url');
   const ial1Attributes = ['email', 'all_emails', 'x509_subject', 'x509_presented', 'verified_at'];
 
   // Functions
   const toggleIAL1Options = () => {
-    ialAttributesCheckboxes.each((idx, attr) => {
-      const element = $(attr).find('input');
-
-      if (!ial1Attributes.includes(element.val())) {
-        $(attr).hide();
-        element.prop('checked', false);
+    ialAttributesCheckboxes.forEach(checkboxWrapper => {
+      const checkboxInput = checkboxWrapper.querySelector('input');
+      if (ial1Attributes.includes(checkboxInput.value)) {
+        checkboxWrapper.classList.add('display-none');
+        checkboxInput.checked = false
       }
     });
   };
 
   const toggleIAL2Options = () => {
-    ialAttributesCheckboxes.each((idx, attr) => $(attr).show());
+    ialAttributesCheckboxes.forEach(checkboxWrapper => {
+      checkboxWrapper.classList.remove('display-none');
+    });
+  };
+
+  const toggleIALOptions = (ial) => {
+    switch (ial) {
+      case '1':
+        failureToProofURL.classList.add('display-none');
+        toggleIAL1Options();
+        break;
+      case '2':
+        failureToProofURL.classList.remove('display-none');
+        toggleIAL2Options();
+        break;
+      default:
+        failureToProofURL.remove('display-none');
+        toggleIAL2Options();
+    }
   };
 
+  // Page initialization
+  toggleIALOptions(ialLevel.value);
+
+  // Event trigger
+  ialLevel.addEventListener("change", (event) => toggleIALOptions(event.target.value));
+}
+
+function protocolOptionSetup() {
+  const idProtocols = document.querySelectorAll('input[name="service_provider[identity_protocol]"]');
+  const activeIdProtocol = document.querySelector('input[name="service_provider[identity_protocol]"]:checked');
+  const samlFields = document.querySelectorAll('.saml-fields');
+  const oidcFields = document.querySelectorAll('.oidc-fields');
+  const returnToSpUrl = document.querySelector('#service_provider_return_to_sp_url');
+
+  // Functions
   const toggleSAMLOptions = () => {
-    samlFields.show();
-    oidcFields.hide();
+    samlFields.forEach(field => field.classList.remove('display-none'));
```

Comment:
I went with something a little different, but with the same idea behind it. PTAL

Reply:
Looks good 👍  Might think explicit hide/show could work better for singular elements, since you could avoid the extra arrow function in the `forEach` callback, e.g.

```js
const showElement = (element) => element.classList.remove('display-none');

// ...

fields.forEach(showElement);
```